### ISSUE TITLE
Missing caption end tag incorrectly reported as error?

### ISSUE THREAD

**Issue by LJWatson** [validator/validator#593]
The Optional tags sections in the [W3C HTML spec](http://w3c.github.io/html/syntax.html#optional-tags) and [WHATWG standard](https://html.spec.whatwg.org/multipage/syntax.html#optional-tags) both state that the end tag for ```<caption>``` may be omitted unless the element is folowed by whitespace or a comment. The validator returns an error when the ```</caption>``` is missing though.

Test code:
```
<table>
<caption>Foo table
<tr>
<th>This foo</th><th>That foo</th>
</tr>
<tr>
<td>Strong</td><td>Weak</td>
</tr>
</table>
```

Discovered whilst investigating w3c/html#922

**Comment by sideshowbarker** [comment]
Nice catch. The error message in this case comes from the HTML parser the checker uses, and looking back at the spec’s commit history, I see the relevant requirements in the parsing algorithm changed in https://github.com/whatwg/html/commit/759ee62b8b01d9a00a73553aa8740e936e1583a2 in 2014 but we never got around to updating the HTML parser code.

I’ve gone ahead and also filed https://bugzilla.mozilla.org/show_bug.cgi?id=1429415, because Firefox uses the same HTML parser — so this should be fixed in Firefox as well, not just in the HTML checker.

I’ll also write up a patch with a fix and see if I can get some time from @hsivonen to review it.

**Comment by sideshowbarker** [comment]
I’ve gone ahead and merged the fix for this and pushed it to https://validator.w3.org/nu/
Thanks again for having caught this
